system/Config/Factories.php

```diff
@@ -65,11 +68,44 @@ class Factories
      * A multi-dimensional array with components as
      * keys to the array of name-indexed instances.
      *
+     * [component => [FQCN => instance]]
+     *
      * @var array<string, array<string, object>>
      * @phpstan-var  array<string, array<class-string, object>>
      */
     protected static $instances = [];
 
+    /**
+     * Define the class to load. You can *override* the concrete class.
+     *
+     * @param string $component Lowercase, plural component name
+     * @param string $name      Classname. The first parameter of Factories magic method
+     * @param string $classname FQCN to load
```

Comment:
These can get mixed up very easily. Let's stick to something like "shortname" for the unqualified version. 
```suggestion
     * @param string $name      Class shortname. The first parameter of Factories magic method
     * @param string $classname FQCN to load
```

Reply:
Or I see I called it "basename" below. Doesn't matter as long as we are consistent.

Reply:
I introduced a new term "class alias".